warehouse/http.py

```diff
@@ -0,0 +1,27 @@
+import threading
+import requests
+
+
+class ThreadLocalSessionFactory(threading.local):
```

Comment:
This doesn't actually give us what I was after: A persistent, thread local session, it will be a thread local, per-request session (so additional complexity that has no benefit). Will fix.